src/examples/weightedcoin.rs

```diff
@@ -0,0 +1,80 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+ use rurel::mdp::{Agent, State};
+ use rurel::strategy::explore::RandomExploration;
+ use rurel::strategy::learn::QLearning;
+ use rurel::strategy::terminate::SinkStates;
+ use rurel::AgentTrainer;
+
+const TARGET : i32 = 100;
+const WEIGHT : u8 = 100; //portion of 255
+ 
+ #[derive(PartialEq, Eq, Hash, Clone)]
+ struct CoinState{
+    balance : i32
+ }
+
+ #[derive(PartialEq, Eq, Hash, Clone)]
+ struct CoinAction{
+    bet : i32
+ }
+ 
+impl State for CoinState {
+    type A = CoinAction;
+
+    fn reward(&self) -> f64 {
+        if self.balance>=TARGET {1.0}
+        else {0.0}
+    }
+
+    fn actions(&self) -> Vec<CoinAction> {
+        let bet_range = {
```

Comment:
Can you help me understand this bit? Why is the range not always `1..self.balance+1`?

Reply:
Winning a bet that's more than TARGET-self.balance would put the player with more balance than the target amount, which is functionally identical to having exactly the target amount. Losing that bet would put the player at a lower balance than losing the smaller bet. Therefore, the smaller bet is strictly better. Learning is faster when there are fewer options to consider at each state.

Reply:
Thanks!